lib/vega_lite/data.ex

```diff
@@ -327,6 +327,311 @@ defmodule VegaLite.Data do
+    * `mark type` - a valid mark to be passed into `VegaLite.mark/3`
+    * `mark opts` - a keyword to be passed into `VegaLite.mark/3`
+
+  ## Options
+
+    * `:radius_marks` - optional list of values for the radius markings. If not passed,
+      will be inferred from the given data points.
+    * `:legend_key` - optional string for accessing `data points` for grouping in the plot legend.
+    * `:direction` - one of `:counter_clockwise` or `:clockwise`, which will affect the direction
+      in which the angles grow from the starting point. Defaults to `:counter_clockwise`.
+    * `:angle_marks` - list of angles for marking the grid. Defaults to `Enum.to_list(0..360//90)`.
+    * `:angle_offset` - offset for the 0º reference. The sign will obey the `:direction` option. Defaults to `0`.
+    * `:grid_opacity` - opacity of the grid. Defaults to `1`.
+    * `:grid_color` - backround color for the grid. Defaults to `"white"`.
+    * `:grid_stroke_color` - stroke color for the grid markings. Defaults to `"black"`.
+    * `:grid_stroke_opacity` - opacity for the grid strokes. Defaults to `1`.
+    * `:scheme` - the color scheme for multi-line plots. Defaults to `"turbo"`
+
+  ## Examples
+
+  In the example below, we plot a dataset by specifying stylized point marks and connecting it with
+  an interpolated line.
+
+      data = %{"r" => [1, 2, 3, 3, 4], "theta" => [0, 30, 45, 135, 270]}
+
+      VegaLite.Data.polar_plot(
+        VegaLite.new(title: "Polar Plot", height: 500, width: 500),
+        [
+          {data, :point, [stroke: "white", stroke_width: 3, tooltip: [data: true]]},
+          {data, :line, point: true, interpolate: "cardinal", color: "black", tooltip: [data: true]}
+        ],
+        angle_marks: [0, 15, 30, 45, 60, 90, 180, 270, 360],
+        radius_marks: [0, 1, 2, 3, 4, 5]
+      )
+
+  In the next example, we add a second line to the plot grouping by the `:legend_key`.
+  Note that the `:legend_key` will also be the legend title, so a readable name is desirable.
+
+      legend_key = "Line Groups"
+
+      data = %{
+        "r" => [1, 2, 3, 3, 4],
+        "theta" => [0, 30, 45, 135, 270],
+        legend_key => List.duplicate("First Line", 5)
+      }
+
+      other_data = %{
+        "r" => [8, 6, 4, 2, 0],
+        "theta" => [0, 30, 60, 90, 120],
+        legend_key => List.duplicate("Second Line", 5)
+      }
+
+      VegaLite.Data.polar_plot(
+        VegaLite.new(title: "Polar Plot", height: 500, width: 500),
+        [
+          {data, :point, [stroke: "black", stroke_width: 3]},
+          {data, :line, point: true, interpolate: "cardinal"},
+          {other_data, :point, [stroke: "black", stroke_width: 3]},
+          {other_data, :line, point: true, interpolate: "cardinal"}
+        ],
+        angle_marks: [0, 15, 30, 45, 60, 90, 180, 270, 360],
+        legend_key: legend_key
+      )
+  """
+  def polar_grid(vl \\ Vl.new(), opts \\ []) do
```

Comment:
@josevalim @cristineguadelupe I was able to split `polar_grid` and `polar_plot` (docs still pending, I'll leave this for later).

The idea is that `polar_grid` will just generate an empty polar grid, while `polar_plot` will convert polar coordinates data into a cartesian plot.

Both can be combined to plot multiple datasets onto a given grid.

I also implemented the `append_layers` `defp` which we might want to extract.

![Screenshot 2023-08-19 at 9 53 21 AM](https://github.com/livebook-dev/vega_lite/assets/16843419/141d47db-26ed-45df-b884-6fd252f61410)
![Screenshot 2023-08-19 at 9 53 24 AM](https://github.com/livebook-dev/vega_lite/assets/16843419/77772698-b4f6-46a3-b2a1-a87a87f386bd)

Reply:
Forgot to show just vl_grid:

![Screenshot 2023-08-19 at 9 56 27 AM](https://github.com/livebook-dev/vega_lite/assets/16843419/558d3ab2-6a51-472c-aea0-c2f1e39bc5ba)

Reply:
This looks pretty good. I think we are on a path to merging this 🎉